controllers/managedcluster_controller.go

```diff
@@ -0,0 +1,123 @@
+package controllers
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/utils"
+	viewv1beta1 "github.com/stolostron/multicloud-operators-foundation/pkg/apis/view/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type ManagedClusterReconciler struct {
+	Client client.Client
+	Scheme *runtime.Scheme
+	Logger *slog.Logger
+}
+
+const odfConfigMapName = "odf-info"
+
+func (r *ManagedClusterReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger := r.Logger.With("ManagedCluster", req.NamespacedName)
+	logger.Info("Reconciling ManagedCluster")
+
+	var managedCluster clusterv1.ManagedCluster
+	if err := r.Client.Get(ctx, req.NamespacedName, &managedCluster); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			logger.Error("Failed to get ManagedCluster", "error", err)
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if err := r.ensureManagedClusterViews(ctx, managedCluster.Name); err != nil {
+		logger.Error("Failed to ensure ManagedClusterView", "error", err)
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Successfully reconciled ManagedCluster", "name", managedCluster.Name)
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ManagedClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.Logger.Info("Setting up ManagedClusterReconciler with manager")
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&clusterv1.ManagedCluster{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(
+			&viewv1beta1.ManagedClusterView{},
+			handler.EnqueueRequestsFromMapFunc(r.managedClusterViewRequestMapper),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}, predicate.GenerationChangedPredicate{}),
+		).
+		Complete(r)
+}
+
+func (r *ManagedClusterReconciler) managedClusterViewRequestMapper(ctx context.Context, obj client.Object) []reconcile.Request {
```

Comment:
This cannot be done as EnqueueRequestsFromMapFunc expects a `type MapFunc func(context.Context, client.Object) []reconcile.Request` 

I would like to avoid wrapping it up in another func :)

Reply:
For simplicity, I prefer removing the logging from this function and moving this function inside `SetupWithManager` function. This keeps things readable and in context.
And you can be assured that the received object is a ManagedClusterView object unless there is a bug in platform. In rare case that it does happen, we can just skip it.